tests/test_calendar_analyzer.py

```
@@ -86,9 +98,15 @@
 
 def test_end_date_before_start_date(monkeypatch, capsys):
     """Test that end date before start date causes system exit."""
+    # Create a temporary dummy file path (secure alternative to mktemp)
+    import tempfile
+    dummy_file = tempfile.NamedTemporaryFile(suffix=".ics", delete=False)
```

Ruff (0.11.9)

Use a context manager for opening files

(SIM115)